app/routes.py

```diff
@@ -1,2 +1,338 @@
+        #application that work with other libraries that work with website
+
+    elif request.method == "POST":
+        request_body = request.get_json()
+        if "title" not in request_body:
+            # return make_response("400 Bad Request")
+            return {"details": "Invalid data"}, 400
+
+        new_goal = Goal(
+            title= request_body["title"]
+        )
+        db.session.add(new_goal)
+        db.session.commit()
+
+        return {
+            "goal": {
+                "id": new_goal.goal_id,
+                "title": new_goal.title
+            }
+        }, 201
+
+@goals_bp.route("/<goal_id>", methods= ["GET", "PUT", "DELETE"])
+def handle_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    if request.method == "GET":
+        if goal is None:
+            return jsonify(None), 404
+        else:
+             return {
+                "goal": {
+                    "id": goal.goal_id,
+                    "title": goal.title
+                }
+            }
+
+    elif request.method == "PUT":
+        if goal is None:
+            return jsonify(None), 404
+        else:
+            form_data = request.get_json()
+
+            goal.title = form_data["title"]
+
+            db.session.commit()
+
+        # return make_response(f"Book #{book.id} successfully updated")
+            return {
+                    "goal": {
+                        "id": goal.goal_id,
+                        "title": goal.title
+                    }
+                }
+   
+    if request.method == "DELETE":
+        if goal is None:
+            return jsonify(None), 404
+        # request_body = request.get_json() request body contains information to help conduct the actual request
+        # ie post request requires data to determines how to actually create the task
+        db.session.delete(goal)
+        db.session.commit()
+
+        return {"details": f'Goal {goal.goal_id} "{goal.title}" successfully deleted'}
+
+@goals_bp.route("/<goal_id>/tasks", methods= ["GET", "POST"])
+def handle_goal_tasks(goal_id):
+#we always pass goal id because every task is related to a goal
+#the goal is the one, tasks are the many
+
+    goal = Goal.query.get(goal_id)
+    if goal is None:
+        # return "Author not found", 404
+        return jsonify(None), 404
+
+    if request.method == "GET":
+        tasks_response = []
+        for task in goal.tasks:
+            tasks_response.append(
+                {
+                "id": task.task_id, #references variable from line 286
+                "goal_id": task.goal_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": False if task.completed_at == None else True
+                }
+            )
+        goal_response = {
+            "id": goal.goal_id,
+            "title": goal.title,
+            "tasks": tasks_response
+        }
+    
```

Comment:
List comprehension and class helper methods in the Goal and Task class would significantly reduce the amount of code for this section.